core/templates/pages/exploration-player-page/services/exploration-player-state.service.ts

```diff
@@ -64,8 +64,22 @@ angular.module('oppia').factory('ExplorationPlayerStateService', [
       ReadOnlyExplorationBackendApiService, StateClassifierMappingService,
       StatsReportingService, UrlService,
       EXPLORATION_MODE) {
-    StatsReportingService = (
-      OppiaAngularRootComponent.statsReportingService);
+    // Due to the migration path we've taken to upgrade AngularJS to Angular 8,
+    // the instances of Angular 8 services are bifurcated; the instance provided
+    // to AngularJS services is different than the one provided to Angular 8
+    // services.
+    //
+    // PlaythroughService and StatsReportingService are both stateful, however,
+    // so it is cruicial for their instances to stay consistent across all
+    // services. To patch the issue described in the preceding paragraph, we
+    // import them directly from a central Angular 8 service,
+    // OppiaAngularRootComponent, to ensure that only the Angular 8 instances
+    // get referenced.
+    //
+    // TODO(#7222): Stop doing this once we've upgraded everything to Angular 8.
+    PlaythroughService = OppiaAngularRootComponent.playthroughService;
+    StatsReportingService = OppiaAngularRootComponent.statsReportingService;
```

Comment:
Should we remove DIs of these services in this directive and remove the corresponding require statements too since we don't need them?

Reply:
@kevintab95, this might not be the best thing to do. While what you have asked correct but this change moves the code away from the "optimal solution". The best solution (at least in my opinion) is this: https://github.com/oppia/oppia/pull/10714#pullrequestreview-491686154

Reply:
I see. In that case, I would suggest the following:
1. Remove the DI and require because there is no use of it.
2. Create an issue to track the optimal solution and mention that in the TODO so that it can be fixed in a follow-on PR.

Reply:
Ah I see. I'm fine with this if it makes things easier from the "angular migration" perspective. Thanks!

But.. please make sure to track the "optimal solution" that @srijanreddy98 has described inhttps://github.com/oppia/oppia/pull/10714#pullrequestreview-491686154.

Reply:
Resolved!